File: ex2_logistic_regression/plotData.m
function plotData(X, y)
 
%PLOTDATA Наносит точки данных X и Y на новую фигуру
% PLOTDATA (x, y) отображает точки данных с `+` для положительных примеров
% и `о` для отрицательных примеров. Предполагается, что X является матрицей Mx2.

% Create New Figure
figure; hold on;

% ====================== YOUR CODE HERE ======================
% Инструкции: нанесите положительные и отрицательные примеры на
% 2D-график, используя опцию 'k+' для положительного
% examples и 'ko' для отрицательных примеров.
%



% format: plot (X, Y, PROPERTY, VALUE, ...)
% Positive values should be black/cross:
%    markerstyle: `+'  crosshair
%    color:       `k'  blacK
% Negative values should be yellow/circle:
%    markerstyle: `o'  circle
%    color:       ‘y’  yellow

% First find Indices of Positive and Negative Examples
pos = find(y == 1); % поступил
neg = find(y == 0); % не поступил

% Plot Postive Result: Черные точки
plot(X(pos, 1), X(pos, 2), 'k+','LineWidth', 2, 'MarkerSize', 7);

% Plot Negative Result: Полые точки
plot(X(neg, 1), X(neg, 2), 'ko', 'MarkerFaceColor', 'y', 'MarkerSize', 7);






% =========================================================================



hold off;

end

File: PythonExample/NumPy/ex2/plotData.py
import numpy as np
import matplotlib.pyplot as plt

'''
function plotData(X, y)
%PLOTDATA Plots the data points X and y into a new figure 
%   PLOTDATA(x,y) plots the data points with + for the positive examples
%   and o for the negative examples. X is assumed to be a Mx2 matrix.

% Create New Figure
figure; hold on;

% ====================== YOUR CODE HERE ======================
% Instructions: Plot the positive and negative examples on a
%               2D plot, using the option 'k+' for the positive
%               examples and 'ko' for the negative examples.
%

    pos = find(y==1); 
    neg = find(y == 0);
    
    plot(X(pos, 1), X(pos, 2), 'k+','LineWidth', 2, 'MarkerSize', 7);
    plot(X(neg, 1), X(neg, 2), 'ko', 'MarkerFaceColor', 'y','MarkerSize', 7);

% =========================================================================



hold off;

end
'''


# noinspection PyPep8Naming
def plot_data(X, y):
    neg = np.where(y == 0)[0]
    pos = np.where(y == 1)[0]

    plt.figure()
    plt.scatter(X[neg, 0], X[neg, 1], c='r', marker='o')
    plt.scatter(X[pos, 0], X[pos, 1], c='b', marker='+')

    return plt
